app/product.go

```diff
@@ -31,6 +31,9 @@ func (s *Server) initializeProducts(
 	// create a product map to consume
 	pmap := make(map[string]struct{})
 	for name := range productMap {
+		if s.blockProduct(name) {
```

Comment:
Nit: I feel like using the word "block" might mean "wait indefinitely". A better word might be "shouldStart".